### ISSUE TITLE
What player modifications make a difference to the seed results?

### ISSUE THREAD

**Issue by friendlytherapist** [KaptainWutax/SeedCracker#36]
Is it possible to crack the seed by disabling everything but dungeons? I have tried this and it outputs about 7 structure seeds and then what do I do with them. Im a bit confused on how and what the mod needs to gather a correct seed. For instance, does the biome setting matter if players modify the blocks? What about the shipwrecks? If I break a wood plank in a shipwreck and enable shipwrecks in the settings will I not get a seed? Im wondering if you could give more explanation on what player changes make a difference to the seed cracking results. Since 99% of the time the mod is used to find seeds on servers that will always be player modified in some ways.

**Comment by Neffc** [comment]
Yes, it is possible to find the seed using just dungeons. I've tested this using only the dungeon finder on a 1.15.2 paper/spigot server. When the cracker outputs the several potential (dungeon?) seeds, you need additional information to calculate which one is correct.

In my experience (and I'm not sure if this is or isn't necessary), going to another dimension, End or Nether, fetches the hashed seed and quickly calculates and outputs the world seed of you've already done the dungeon cracking part.

**Comment by friendlytherapist** [comment]
Did you disable everything but dungeons?

**Comment by Neffc** [comment]
> I've tested this using only the dungeon finder on a 1.15.2 paper/spigot server.
Yes.

**Comment by friendlytherapist** [comment]
It works thanks. Its not as easy as it looks. Since clearing all seed data also clears the worlds hashed seed and resets categories to default so what you have to do to get it back is go to another dimension in the world. Nether works best. Heres what I did:

log on to server hub.
Switch to the survival server.
Go to nether dimension through vanilla means.
Clear seed data in nether.
Set everything to false through categories.
Enable dungeons through types.
Go back to overworld.
You should see a get hashed seed notification in chat.
Gather a few dungeons. (32+ bits)
You should now see a hashed decorators seeds notification.
It then works its magic and lists the seed in chat.

**Comment by Neffc** [comment]
Glad to hear it worked for you. I opted to set the other finders off by default (with only the dungeon finder on) in the profile before compiling. This means you don't have to worry about clearing data if you know some of the other decorators are incorrect/modified.

**Comment by KaptainWutax** [comment]
> For instance, does the biome setting matter if players modify the blocks? What about the shipwrecks? If I break a wood plank in a shipwreck and enable shipwrecks in the settings will I not get a seed? Im wondering if you could give more explanation on what player changes make a difference to the seed cracking results. Since 99% of the time the mod is used to find seeds on servers that will always be player modified in some ways.

It's a bit complicated but it should NEVER throw false positives. Breaking a piece of wood in a shipwreck probably won't affect it since the structure can still be uniquely identified. It depends on which wood block though. Overall, each structure has it's own restrictions and I tried really hard to make sure bad data never gets collected.

**Comment by friendlytherapist** [comment]
Thanks, the dungeon cracker works well but I havent had any luck with end decorator cracking.

**Comment by KaptainWutax** [comment]
End decorator cracking?

**Comment by friendlytherapist** [comment]
Using the end pillars and end gateways to complete the seed

**Comment by KaptainWutax** [comment]
I don't understand what you're referring to. The end features work the same as everything else.

**Comment by friendlytherapist** [comment]
If youre cracking without dungeon finder, you need the end dimension with gateways and pillars to find the seed right? What im saying is the non dungeon cracking never worked for me and I suspect its because my servers end dimension is wrong in some way. I think some of the end gateways are off by one block or something.
